addons/mail/static/src/models/thread/thread.js

```diff
@@ -1464,7 +1464,12 @@ function factory(dependencies) {
         * @returns {boolean}
         */
         _computeIsChannelDescriptionChangeable() {
-            return this.model === 'mail.channel' && ['channel', 'group'].includes(this.channel_type);
+            return (
+                this.model === 'mail.channel' &&
+                ['channel', 'group'].includes(this.channel_type) &&
+                this.messaging.currentUser &&
+                this.messaging.currentUser.isInternalUser
+            );
```

Comment:
The name of this becomes "slightly incorrect" because it now means `isChannelDescriptionChangeableByCurrentUser`.

I would suggest in stable to keep the current field, and create a new field for this new purpose.